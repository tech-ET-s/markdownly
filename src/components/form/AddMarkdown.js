import React from 'react';
import PropTypes from 'prop-types';

function AddMarkdown({ formInputSubmit, formInputChange, formInput }) {
  return (
    <form onSubmit={event => formInputSubmit(event, formInput)}>
      <input value={formInput} onChange={formInputChange}/>
      <button>+</button>
    </form>
  );
}

AddMarkdown.propTypes = {
  formInput: PropTypes.string.isRequired,
  formInputSubmit: PropTypes.func.isRequired,
  formInputChange: PropTypes.func.isRequired
};

export default AddMarkdown;
