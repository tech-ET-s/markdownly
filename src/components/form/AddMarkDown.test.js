import React from 'react';
import { shallow } from 'enzyme';
import AddMarkdown from './AddMarkdown';

describe('AddMarkdown component', () => {
  it('form to save a markdown obj', () => {
    const wrapper = shallow(<AddMarkdown
      formInputSubmit={()=>{}}
      formInputChange={()=>{}}
      formInput='Hellow'/>);
    expect(wrapper).toMatchSnapshot();
  });
});
