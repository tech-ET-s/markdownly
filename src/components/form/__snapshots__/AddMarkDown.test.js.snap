// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMarkdown component form to save a markdown obj 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    value="Hellow"
  />
  <button>
    +
  </button>
</form>
`;
